test/event.js

```diff
@@ -1,16 +1,91 @@
 import Vue from 'vue';
 
-// Triggers an event that does not bubble.
-export const trigger = (eventName, wrapper, selector = undefined) => {
-  const target = selector == null ? wrapper : wrapper.first(selector);
-  target.trigger(eventName);
-  const nextTick = Vue.nextTick();
-  return selector == null ? nextTick : nextTick.then(() => wrapper);
+
+
+////////////////////////////////////////////////////////////////////////////////
+// CREATING EVENTS
+
+export const dataTransfer = (files) => {
+  const dt = new DataTransfer();
+  for (const file of files) {
+    // DataTransferItemList is not supported in IE. MDN indicates that
+    // DataTransferItemList.prototype.add() is not supported even in Chrome, but
+    // another source seems to imply that it is, and I am not encountering any
+    // issues using it in Headless Chrome.
+    dt.items.add(file);
+  }
+  return dt;
 };
 
-const eventNames = ['change', 'click', 'submit'];
-for (const name of eventNames)
-  trigger[name] = (wrapper, selector = undefined) => trigger(name, wrapper, selector);
+const dragEvent = (type, { target, files, ie = false }) => {
+  const dt = !ie
+    ? dataTransfer(files)
+    /* This object does not have all the properties of a DataTransfer object,
+    and its `files` property is an Array, not a FileList. However, I think the
+    only code that will use the object will be jQuery and Central: I think the
+    object will never actually be passed to the browser's drag functionality. We
+    can add more properties to the object as we need them for testing. */
+    : { files, types: ['Files'], dropEffect: 'none' };
+  const originalEvent = $.Event(type, { dataTransfer: dt });
+  return $.Event(type, { target, originalEvent });
+};
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// trigger
+
+export const trigger = {};
+
+const simpleEventNames = ['change', 'click', 'submit'];
+for (const eventName of simpleEventNames) {
+  // Triggers an event that does not bubble.
```

Comment:
i know this isn't new code but i kind of feel like you should _always_ bubble, rather than never, since that's what the browser will do..

Reply:
I added that comment because it was looking to me like the Vue testing library that we're using, avoriaz, doesn't ever bubble events. In the code, I just call the avoriaz `trigger()` method (the method doesn't have any option for bubbling vs. not), but I thought I was seeing no bubbling. However, looking back at it, the issue I was running into might have been due to Vue not knowing about jQuery's listeners. Maybe I should mention avoriaz specifically in the comment, or maybe I should just remove it if I'm no longer sure that it's right (or link to this thread?).